src/main/scala/com/imageintelligence/http4c/middleware/JWTAuthMiddleware.scala

```diff
@@ -0,0 +1,50 @@
+package com.imageintelligence.http4c.middleware
+
+import com.auth0.jwt.JWT
+import com.auth0.jwt.algorithms.Algorithm
+import com.auth0.jwt.exceptions.JWTVerificationException
+import com.auth0.jwt.interfaces.DecodedJWT
+import org.http4s._
+import org.http4s.dsl._
+import org.http4s.headers.Authorization
+import org.http4s.server.AuthMiddleware
+
+import scalaz._
+import Scalaz._
+import scalaz.concurrent.Task
+
+object JWTAuthMiddleware {
+
+  def getBearerToken(req: Request): String \/ String = {
```

Comment:
yes, but more fundamentally the left hand string represents some sort of failure and the right hand string is the token as a string. Once you have it in the Either monad, you can chain the operations and gather any errors together, as is done in the for comprehension in apply.